vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/GridPro.java

```diff
@@ -289,7 +289,13 @@ public void setValueProvider(ValueProvider<T, ?> valueProvider) {
         String columnId = createColumnId(false);
 
         EditColumn<E> column = this.addColumn((new ColumnComponentPathRenderer<>(columnId,
-                value -> valueProvider.apply(value), renderer)), this::createEditColumn);
+                value -> {
+                    if (valueProvider.apply(value) != null) {
+                        return valueProvider.apply(value).toString();
```

Comment:
`valueProvider.apply()` should be called only once. For most cases it's a simple getter, but in theory it might be a more expensive operation, or it's even possible that it returns a different value on different calls.

Reply:
Stored as `Object item`